kafka-connect-s3/src/test/java/io/confluent/connect/s3/TopicPartitionWriterTest.java

```diff
@@ -624,6 +626,216 @@ public void testWallclockUsesBatchTimePartitionBoundary() throws Exception {
+            .build();
+
+
+    List<Struct> records2 = createRecordBatches(schema2, 3, 6);
+
+    Schema schema3 = SchemaBuilder.struct()
+            .name("record2")
+            .version(1)
+            .field("boolean", Schema.BOOLEAN_SCHEMA)
+            .field("int", Schema.INT32_SCHEMA)
+            .field("long", Schema.INT64_SCHEMA)
+            .field("float", Schema.FLOAT32_SCHEMA)
+            .field("double", Schema.FLOAT64_SCHEMA)
+            .build();
+
+
+    List<Struct> records3 = createRecordBatches(schema3, 3, 6);
+
+    int offset = 0;
+    for (int i=0; i<records1.size(); i++) {
+      topicPartitionWriter.buffer(
+              new SinkRecord(TOPIC, PARTITION, Schema.STRING_SCHEMA, key, schema1, records1.get(i),
+                      offset++));
+      topicPartitionWriter.buffer(
+              new SinkRecord(TOPIC, PARTITION, Schema.STRING_SCHEMA, key, schema2, records2.get(i),
+                      offset++));
+      topicPartitionWriter.buffer(
+              new SinkRecord(TOPIC, PARTITION, Schema.STRING_SCHEMA, key, schema3, records3.get(i),
+                      offset++));
+    }
+    // Test actual write
+    topicPartitionWriter.write();
+    topicPartitionWriter.close();
+
+    String dirPrefix1 = partitioner.generatePartitionedPath(TOPIC,
+            "schema_name" + "=" + "null" + "_partition=" + PARTITION);
+    String dirPrefix2 = partitioner.generatePartitionedPath(TOPIC,
+            "schema_name" + "=" + "record1" + "_partition=" + PARTITION);
+    String dirPrefix3 = partitioner.generatePartitionedPath(TOPIC,
+            "schema_name" + "=" + "record2" + "_partition=" + PARTITION);
+
+    List<Struct> expectedRecords = new ArrayList<>();
+    int ibase = 16;
+    float fbase = 12.2f;
+    // The expected sequence of records is constructed taking into account that sorting of files occurs in verify
+    for (int i = 0; i < 6; ++i) {
+      for (int j = 0; j < 3; ++j) {
+        expectedRecords.add(createRecord(schema1, ibase + j, fbase + j));
+      }
+    }
+    for (int i = 0; i < 6; ++i) {
+      for (int j = 0; j < 3; ++j) {
+        expectedRecords.add(createRecord(schema2, ibase + j, fbase + j));
+      }
+    }
+    for (int i = 0; i < 6; ++i) {
+      for (int j = 0; j < 3; ++j) {
+        expectedRecords.add(createRecord(schema3, ibase + j, fbase + j));
+      }
+    }
+
+    List<String> expectedFiles = new ArrayList<>();
+    for(int i = 0; i < 54; i += 9) {
+      expectedFiles.add(FileUtils.fileKeyToCommit(topicsDir, dirPrefix1, TOPIC_PARTITION, i, extension, ZERO_PAD_FMT));
+      expectedFiles.add(FileUtils.fileKeyToCommit(topicsDir, dirPrefix2, TOPIC_PARTITION, i+1, extension, ZERO_PAD_FMT));
+      expectedFiles.add(FileUtils.fileKeyToCommit(topicsDir, dirPrefix3, TOPIC_PARTITION, i+2, extension, ZERO_PAD_FMT));
+    }
+
+    verify(expectedFiles, 3, schema1, expectedRecords);
+  }
+
+  @Test
+  public void testWriteSchemaPartitionerWithSuffix() throws Exception {
```

Comment:
nit - 
we can refactor common code out of these tests, a config and dirPrefix are the only differences.